Add intro tutorial which pulls content from interactive examples (#1374)
enthought/traits: docs/source/traits_tutorial/introduction.rst

============
Introduction
============

> This is tutorial is intended to introduce you to the basics of Traits

Reviewer: ```suggestion
This tutorial is intended to introduce you to the basics of Traits
```

> and to give you an idea of what capabilities the Traits libraries provide.

Reviewer: ```suggestion
and to give you an idea of what capabilities the Traits library provides.
```

This tutorial assumes that you are confortable with the Python
programming language, object oriented programming, and the core tools of

> the Python scientifc ecosystem, such as NumPy.

Reviewer: ```suggestion
the Python scientific ecosystem, such as NumPy.
```
